aws/resource_aws_transfer_user.go

```diff
@@ -32,14 +32,46 @@ func resourceAwsTransferUser() *schema.Resource {
 			},
 
 			"home_directory": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				ValidateFunc: validation.StringLenBetween(0, 1024),
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"home_directory_mappings"},
+				ValidateFunc:  validation.StringLenBetween(0, 1024),
+			},
+
+			"home_directory_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					transfer.HomeDirectoryTypeLogical,
+					transfer.HomeDirectoryTypePath,
+				}, false),
+			},
+
+			"home_directory_mappings": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      50,
+				ConflictsWith: []string{"home_directory", "policy"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"entry": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1024),
+						},
+						"target": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1024),
+						},
+					},
+				},
 			},
 
 			"policy": {
 				Type:             schema.TypeString,
 				Optional:         true,
+				ConflictsWith:    []string{"home_directory_mappings"},
```

Comment:
This is not a true conflict. AWS SDK Allows for policy along with the home directories.